### PR TITLE
Add partitioning APIs to pylibcudf

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Contributes to https://github.com/rapidsai/cudf/issues/15162

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/pylibcudf/pylibcudf/tests/test_partitioning.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        plc.interop.from_arrow(pa_table),
+        plc.interop.from_arrow(pa.array([0, 0, 0])),
+        1,
+    )
+    pa_result = plc.interop.to_arrow(plc_result)
+    pa_expected = pa.table(
+        [[1, 2, 3], [1, 2, 5], [1, 2, 10]],
+        schema=pa.schema([pa.field("", pa.int64(), nullable=False)] * 3),
+    )
+    assert pa_result.equals(pa_expected)
```
Comment (vyasr) [inline python/pylibcudf/pylibcudf/tests/test_partitioning.py:24]: Is there a reason you aren't using the utilities we have for this like [`assert_column_eq`](https://github.com/rapidsai/cudf/blob/branch-24.10/python/pylibcudf/pylibcudf/tests/common/utils.py#L44)? You shouldn't need to do the arrow conversions manually. Not sure if this applies to other PRs that you currently have open as well. If it does, could you update the others as well.

File: python/pylibcudf/pylibcudf/tests/test_partitioning.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+@pytest.fixture
+def pa_table():
+    return pa.table({"a": [1, 2, 3], "b": [1, 2, 5], "c": [1, 2, 10]})
+
+
+def test_partition(pa_table):
+    plc_result, result_offsets = plc.partitioning.partition(
+        plc.interop.from_arrow(pa_table),
```
Comment (vyasr) [inline python/pylibcudf/pylibcudf/tests/test_partitioning.py:15]: We generally prefer putting both the arrow object and the corresponding cudf object into a combined fixture and making it have module scope. That minimizes the number of distinct GPU allocations required to run the test suite. [Here's an example](https://github.com/rapidsai/cudf/blob/branch-24.10/python/pylibcudf/pylibcudf/tests/test_reshape.py#L9). [Some older tests](https://github.com/rapidsai/cudf/blob/branch-24.10/python/pylibcudf/pylibcudf/tests/test_binaryops.py) will set those up as two separate fixtures, with the [pylibcudf fixture](https://github.com/rapidsai/cudf/blob/branch-24.10/python/pylibcudf/pylibcudf/tests/test_binaryops.py#L71) depending on the [pyarrow fixture](https://github.com/rapidsai/cudf/blob/branch-24.10/python/pylibcudf/pylibcudf/tests/test_binaryops.py#L64). That is also OK, but overall the combined fixture is a bit cleaner.

**Review by vyasr — CHANGES_REQUESTED** [review]
Some suggestions to improve the tests and align them with existing testing patterns. They may also apply to other PRs that you have open.

**Review by Matt711 — APPROVED** [review]

**Review by vyasr — APPROVED** [review]

**Pull request merged**